nacl-fast.js

```diff
@@ -2361,17 +2361,16 @@ nacl.setPRNG = function(fn) {
   // Initialize PRNG if environment provides CSPRNG.
   // If not, methods calling randombytes will throw.
   var crypto;
-  if (typeof window !== 'undefined') {
-    // Browser.
-    if (window.crypto && window.crypto.getRandomValues) {
-      crypto = window.crypto; // Standard
-    } else if (window.msCrypto && window.msCrypto.getRandomValues) {
-      crypto = window.msCrypto; // Internet Explorer 11+
-    }
-    if (crypto) {
+  if (typeof process === 'undefined' || process.browser) {
```

Comment:
Ah got it. Then I vote just for `process.browser` removal.
On Feb 20, 2016 18:43, "Dmitry Chestnykh" <notifications@github.com> wrote:

> In nacl-fast.js
> <https://github.com/dchest/tweetnacl-js/pull/87#discussion_r53547362>:
>
> > @@ -2361,17 +2361,16 @@ nacl.setPRNG = function(fn) {
> >    // Initialize PRNG if environment provides CSPRNG.
> >    // If not, methods calling randombytes will throw.
> >    var crypto;
> > -  if (typeof window !== 'undefined') {
> > -    // Browser.
> > -    if (window.crypto && window.crypto.getRandomValues) {
> > -      crypto = window.crypto; // Standard
> > -    } else if (window.msCrypto && window.msCrypto.getRandomValues) {
> > -      crypto = window.msCrypto; // Internet Explorer 11+
> > -    }
> > -    if (crypto) {
> > +  if (typeof process === 'undefined' || process.browser) {
>
> Here is my point: native crypto is better than DOM realisation. So check if
> require('crypto') is present and usable, than fallback to DOM window.crypto
>
> No, both are pretty much equally good CSPRNG (window.crypto is implemented
> by browsers natively, Node's crypto is from OpenSSL).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dchest/tweetnacl-js/pull/87/files#r53547362>.
>